src/pages/iou/request/step/IOURequestStepConfirmation.tsx

```diff
@@ -259,6 +264,10 @@ function IOURequestStepConfirmation({
     }, []);
 
     useEffect(() => {
+        if (isCreatingTrackExpense && realPolicyID !== undefined && policy?.pendingAction !== CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD) {
```

Comment:
@parasharrajat There was a previously deleted comment here about using `realPolicyID`.
I think you're right and it's a bit confusing. I haven't noticed that below we have similar condition when creating `policyID` for tracked and other expenses. 
In both cases when isCreatingTrackExpense is true we return `policyForMovingExpensesID`. So I think we should actually pass this IMO instead of `realPolicyID` wdyt?